### DISCUSSION TITLE
basic macro

### DISCUSSION THREAD

**Discussion by arembridge** [nickcoutsos/keymap-editor#204] [Q&A] [answered]
firstly, this tool is absolutely awesome!  zmk isn't the easiest for n00bs (i used via configurator which is super user-friendly) but this gave me the confidence to get nearly everything i wanted in zmk! thanks for putting the time into making and sharing such an awesome tool 😊 

i can't figure out how to bind a key press to execute `option(down) -> left arrow -> option(up)` and similar macros.  

if someone can help explain, i'd be happy to contribute to the README.  thanks!

**Comment by nickcoutsos** [comment, marked as answer]
Hi @arembridge 

That normally doesn't require macros, check out [modifier functions](https://zmk.dev/docs/codes/modifiers#modifier-functions). Under weirder OS/software timing circumstances you may benefit from a macro in which case you can go over to the _Macros_ tab and create one with the appropriate sequence of `&kp ...` and other binds, but let's start with the simple approach.
